resources/install.rb

```diff
@@ -174,10 +174,22 @@
   when 'repo'
     case node['platform_family']
     when 'amazon', 'fedora', 'rhel'
+      uses_dnf = if platform_family?('fedora')
+                   node['platform_version'].to_i > 22
+                 elsif platform_family?('rhel')
+                   node['platform_version'].to_i > 7
+                 else
+                   false
+                 end
       yum_repository 'nginx' do
         description  'Nginx.org Repository'
         baseurl      repo_url
         gpgkey       repo_signing_key
+        notifies :run, 'execute[disable-nginx-module]', :immediately if uses_dnf
+      end
+      execute 'disable-nginx-module' do
+        command 'dnf -y module disable nginx'
+        action :nothing
```

Comment:
I recommend you use the same format I mentioned in postgresql. This also removes the need to notify the resource.